subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -739,7 +757,7 @@ protected boolean isSharedDaemons() {
     @Override
     public boolean isUseDaemon() {
         CliDaemonArgument cliDaemonArgument = resolveCliDaemonArgument();
-        if (cliDaemonArgument == NO_DAEMON) {
+        if (cliDaemonArgument == NO_DAEMON ||cliDaemonArgument == FOREGROUND) {
```

Comment:
Foreground mode is not really a daemon in the sense that this method means. It's fully running in the client, just like with no-daemon, but it is long-lived.